docs/sequence_diagram.uml

```diff
@@ -0,0 +1,56 @@
+//
+// vim:syntax=plantuml
+//
+
+// Generate PNG image with sequence diagram by using the following command:
+// java -jar plantuml.jar sequence_diagram.uml
+//
+// Generate SVG drawing with sequence diagram by using the following command:
+// java -jar plantuml.jar -tsvg sequence_diagram.uml
+
+@startuml
+
+header Sequence diagram for Lightspeed Service
+footer Copyright © 2024 Red Hat, Inc. Author: Pavel Tisnovsky
+
+actor "User" as user
+participant "Lightspeed\nService" as ols
+control "Question\nValidator" as question_validator
+control "Document\nSummarizer" as document_summarizer
+database "Conversation cache" as cache
+collections "LLM" as llm
+
+== Question ==
+
+user -> ols: Question
+
+== Conversation history ==
+
+ols -> cache: Retrieve conversation history
+cache -> ols: Conversation history
+
+== Validation ==
+
+ols -> question_validator: Validate question
+question_validator -> llm: Query
+llm -> question_validator: Valid or invalid
+
+== Answer generation ==
+
+alt "Invalid question case"
+question_validator -> ols: Invalid question
+ols -> user: Your question is not valid
+end
+
+alt "Valid question case"
+question_validator -> ols: Valid question
+ols -> document_summarizer: Summarize document
+document_summarizer -> llm: Query
+llm -> document_summarizer: Answer
```

Comment:
How about also showing the RAG retrieval piece of the flow?  You mentioned the docs retriever as a component but then didn't include it in the flow diagram

Note that rag retrieval is currently "optional" (if there's no RAG content configured, we skip the retrieval step but still send the user query to the llm)

Reply:
yup, I'll add it

Reply:
updated, but not sure about naming @bparees

Reply:
good enough for now, thanks